provider/cluster/kube/inventory.go

```diff
@@ -478,6 +478,11 @@ func (c *client) nodeIsActive(node corev1.Node) bool {
 		}
 	}
 
+	// If the node has been tainted, don't consider it active.
+	if len(node.Spec.Taints) != 0 {
```

Comment:
I think we need to iterate through the list and only consider taints that have an effect of `NoSchedule` or `NoExecute`